src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,350 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Text;
+using JetBrains.Annotations;
+using NLog.Common;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog.Layouts
+{
+    /// <summary>
+    /// Layout with a simple value (e.g. int) or a layout which results in a simple value (e.g. ${counter})
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class Layout<T> : Layout, IRenderable<T>
+    {
+        // ReSharper disable StaticMemberInGenericType - this is safe, static ctor is called for every generic type
+        [NotNull] private static readonly Type Type;
+        [NotNull] private static readonly string TypeNamed;
+        // ReSharper restore StaticMemberInGenericType
+
+        [CanBeNull] private readonly Layout _layout;
+        private readonly T _fixedValue;
+
+        /// <inheritdoc />
+        public Layout(T value)
+        {
+            _fixedValue = value;
+            IsFixed = true;
+            _layout = null;
+        }
+
+        /// <inheritdoc />
+        public Layout(Layout layout)
+        {
+            IsFixed = TryGetFixedValue(layout, out _fixedValue);
+            _layout = layout;
+        }
+
+        static Layout()
+        {
+            var type = typeof(T);
+            Type = Nullable.GetUnderlyingType(type) ?? type;
+            TypeNamed = type.Name;
+        }
+
+        /// <summary>
+        /// Is fixed value?
+        /// </summary>
+        public bool IsFixed { get; }
+
+        #region Overrides of TypedLayout<T>
+
+        private static string ValueToString(T value, CultureInfo cultureInfo)
+        {
+            if (value is IConvertible convertible)
+            {
+                return convertible.ToString(cultureInfo);
+            }
+
+            return value?.ToString();
+        }
+
+        private bool TryParse(string text, out T value, T defaultValue)
+        {
+            return TryConvertTo(text, out value, defaultValue);
+        }
+
+        private bool TryConvertTo(object raw, out T value, T defaultValue)
+        {
+            if (raw == null)
+            {
+                value = defaultValue;
+                return false;
+            }
+
+            var cultureInfo = CultureInfo.CurrentCulture;
+            try
+            {
+                var converter = ResolveService<IPropertyTypeConverter>() ?? PropertyTypeConverter.Instance;
+                var convertedValue = converter.Convert(raw, Type, null, cultureInfo);
+                if (convertedValue is T goodValue)
+                {
+                    value = goodValue;
+                    return true;
+                }
+            }
+            catch (Exception e)
+            {
+                InternalLogger.Debug(e, "Conversion to type {0} failed", Type);
```

Comment:
Maybe include the output of `raw`-value in the debug-output

Reply:
I was afraid it could be large. But maybe that isn't important?

Reply:
Not when it fails.

Reply:
Will do otherwise a "left 1000" or something like that